torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -622,13 +622,16 @@ def inner(*flat_args):
             output_info.append(out_info)
 
         # See Note [AOT Autograd: Views to avoid tangents aliasing inputs]
-        def view_avoid_dupes_with_primals(t):
+        def view_avoid_dupes_with_primals(t, parent_cls, attr):
             if isinstance(t, Tensor) and is_traceable_wrapper_subclass(t):
                 return transform_subclass(
-                    t, lambda _, inner_t: view_avoid_dupes_with_primals(inner_t)
+                    t, lambda attr, inner_t: view_avoid_dupes_with_primals(inner_t, type(t), attr)
                 )
             if isinstance(t, Tensor):
-                return t.view(t.shape)
+                out = t.view(t.shape)
+                if parent_cls is torch.nested._internal.nested_tensor.NestedTensor and (attr == "_offsets" or attr == "_lengths"):
+                    out.set_nested_int(t.nested_int())
```

Comment:
```suggestion
                if hasattr(t, "nested_int"):
                    out.set_nested_int(t.nested_int())
```

maybe? to keep it slightly cleaner